design.md

```diff
@@ -39,16 +48,43 @@
   - Kinematics (e.g., bases, limbs, links, DH parameters, root offsets).
   - Scripts (e.g., body/limb/joint controllers, body/limb CAD generators).
 
+### Plugins
+
+- The IDE must be able to display a searchable list of plugins.
+  - The IDE must display the group, name, and version triple.
+  - The IDe must display the plugin artifact's description.
+  - By default, the IDE displays all plugins in the group `com.commonwealthrobotics`.
+  - The user may filter for a specific type of plugin (e.g. device, toolchain, device resource, or simulator).
+- There is a list of default plugins which are installed by default.
+
+### Git and GitHub Integration
+
+- The IDE must present a simplified view of Git and GitHub to the user.
+  - The user must be able to commit and push their changes from one simplified modal interface.
```

Comment:
Is there anything Bowler-specific about the Git UI enhancements? If not, it might be better to produce that as a separate plugin (which is obviously also installed by default in our IntelliJ build). Separation of concerns/UNIX philosophy and whatnot.

Reply:
I would like to only produce one plugin for now to keep engineering costs down.